linera-core/src/client/mod.rs

```diff
@@ -2101,12 +2101,18 @@ where
         &self,
         operations: Vec<Operation>,
         blobs: Vec<Blob>,
+        guard: Option<OwnedMutexGuard<()>>,
     ) -> Result<ExecuteBlockOutcome, ChainClientError> {
         #[cfg(with_metrics)]
         let _latency = metrics::EXECUTE_BLOCK_LATENCY.measure_latency();
 
-        let mutex = self.state().client_mutex();
-        let _guard = mutex.lock_owned().await;
+        let _guard = match guard {
+            Some(guard) => guard,
+            None => {
+                let mutex = self.state().client_mutex();
+                mutex.lock_owned().await
+            }
+        };
```

Comment:
Can you add comment here explaining why we acquire a different lock if `guard` is None?

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/3822/commits/c93bbbe8d391bd81fe128f567830d9b46470d819.